### ISSUE TITLE
exp doesn't handle NaNs

### ISSUE THREAD

**Issue by andreasnoack** [JuliaMath/DoubleFloats.jl#62]
```julia
julia> exp(Double64(NaN))
ERROR: InexactError: Int64(NaN)
Stacktrace:
 [1] Type at ./float.jl:703 [inlined]
 [2] calc_exp(::DoubleFloat{Float64}) at /Users/andreasnoack/.julia/packages/DoubleFloats/FaJUi/src/math/elementary/explog.jl:143
 [3] exp(::DoubleFloat{Float64}) at /Users/andreasnoack/.julia/packages/DoubleFloats/FaJUi/src/math/elementary/explog.jl:18
 [4] top-level scope at none:0
```

**Comment by JeffreySarnoff** [comment]
fixed in v0.7.12 -- just tagged

**Comment by andreasnoack** [comment]
Thanks.
